pkg/customsignaturestore/customsignaturestore.go

```diff
@@ -58,44 +65,56 @@ func (s *Store) Signatures(ctx context.Context, name string, digest string, fn s
 		return done, err
 	}
 
-	stores := make([]store.Store, 0, len(uris))
-	for i := range uris {
-		uri := *uris[i]
-		stores = append(stores, &sigstore.Store{
-			URI:        &uri,
-			HTTPClient: s.HTTPClient,
-		})
+	stores := make([]store.Store, 0, len(customStores))
+	for _, customStore := range customStores {
+		uri, err := url.Parse(customStore.URL)
+		if err != nil {
+			return err
+		}
+		newHttpClient, err := s.HTTPClient(customStore.CA.Name)
+		if err != nil {
+			return err
+		}
+		if customStore.CA.Name != "" {
+			stores = append(stores, &sigstore.Store{
+				URI:        uri,
+				HTTPClient: func() (*http.Client, error) { return newHttpClient, nil }})
+		}
+
+	}
+	if len(stores) == 0 {
+		return errors.New("ClusterVersion spec.signatureStores exhausted without finding a valid signature")
 	}
 	store := &parallel.Store{Stores: stores}
 	if err := store.Signatures(ctx, name, digest, wrapper); err != nil || allDone {
 		return err
 	}
-	return errors.New("ClusterVersion spec.signatureStores exhausted without finding a valid signature.")
+	return nil
 }
 
-func (s *Store) refreshConfiguration(ctx context.Context) ([]*url.URL, error) {
-	config, err := s.Lister.Get(s.Name)
+// refreshConfiguration fetches the current ClusterVersion configuration
+// and saves the information to a local copy
+func (s *Store) refreshConfiguration(ctx context.Context) ([]configv1.SignatureStore, error) {
+	config, err := s.ClusterVersionLister.Get(s.Name)
 	if err != nil {
 		return nil, err
 	}
 
-	var uris []*url.URL
+	//Save configv1.SignatureStore.URL and configv1.SignatureStore.CA information
+	// from configv1.SignatureStore to customStores variable
```

Comment:
Also I think the comment is a bit redundant with the code, I'd consider leaving it out

Reply:
Fixed it in https://github.com/openshift/cluster-version-operator/commit/4ead92af5e0927abf842409850329092802b3579